src/functional/ffront/fbuiltins.py

```diff
@@ -12,19 +12,65 @@
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+from dataclasses import dataclass
+from typing import Optional
+
 from numpy import float32, float64, int32, int64
 
-from functional.common import Field
+from functional.common import Dimension, Field
+from functional.ffront import common_types as ct
+from functional.iterator import runtime
 
 
-__all__ = ["Field", "float32", "float64", "int32", "int64"]
+__all__ = ["Field", "float32", "float64", "int32", "int64", "neighbor_sum"]
 
 TYPE_BUILTINS = [Field, float, float32, float64, int, int32, int64, bool, tuple]
 TYPE_BUILTIN_NAMES = [t.__name__ for t in TYPE_BUILTINS]
 
+
+@dataclass()
+class BuiltInFunction:
+    __gt_type: ct.FunctionType
+
+    def __call__(self, *args, **kwargs):
+        """Act as an empty place holder for the built in function."""
+
+    def __gt_type__(self):
+        return self.__gt_type
+
+
+neighbor_sum = BuiltInFunction(
+    ct.FunctionType(
+        args=[ct.DeferredSymbolType(constraint=ct.FieldType)],
+        kwargs={"axis": ct.ScalarType(kind=ct.ScalarKind.DIMENSION)},
+        returns=ct.DeferredSymbolType(constraint=ct.FieldType),
+    )
+)
+
+
+FUN_BUILTIN_NAMES = ["neighbor_sum"]
+
+
 EXTERNALS_MODULE_NAME = "__externals__"
 MODULE_BUILTIN_NAMES = [EXTERNALS_MODULE_NAME]
 
 ALL_BUILTIN_NAMES = TYPE_BUILTIN_NAMES + MODULE_BUILTIN_NAMES
 
 BUILTINS = {name: globals()[name] for name in __all__}
+
+
+# TODO(ricoh): This should probably be reunified with ``iterator.runtime.Offset``
+# potentially in ``functional.common``, which requires lifting of
+# ``ffront.common_types`` into ``functional``.
+@dataclass(frozen=True)
+class FVOffset(runtime.Offset):
+    source: Optional[Dimension] = None
+    target: Optional[tuple[Dimension, ...]] = None
+
+    def __gt_type__(self):
+        return ct.OffsetType(source=self.source, target=self.target)
+
+
+# TODO(ricoh): same applies as for FVOffset above
+def fvoffset(value, source=None, target=None):
+    return FVOffset(value, source, target)
```

Comment:
Is the function really needed? I think `runtime` implements the same pattern, but I think it is basically useless to define a function which basically calls the constructor with the same arguments.